library/x509_crt.c

```diff
@@ -2578,10 +2579,19 @@ int mbedtls_x509_crt_verify_with_profile( mbedtls_x509_crt *crt,
  *    as that isn't done as part of chain building/verification currently
  *  - builds and verifies the chain
  *  - then calls the callback and merges the flags
+ *
+ * The parameters pairs `trust_ca`, `ca_crl` and `f_ca_cb`, `p_ca_cb`
+ * are mutually exclusive: If `f_ca_cb != NULL`, it will be used by the
+ * verification routine to search for trusted signers, and CRLs will
+ * be disabled. Otherwise, `trust_ca` will be used as the static list
+ * of trusted signers, and `ca_crl` will be use as the static list
+ * of CRLs.
  */
-int mbedtls_x509_crt_verify_restartable( mbedtls_x509_crt *crt,
+static int mbedtls_x509_crt_verify_restartable_cb( mbedtls_x509_crt *crt,
```

Comment:
There are now 2 different callbacks. Even though the verify callback is used with all of the versions, should this be distinguished using `mbedtls_x509_crt_verify_restartable_ca_cb`

Reply:
Name nitpicking: static function usually don't start with `mbedtls_` (also, the name here is not consistent with the commit message). But my real issue here is what @jarlamsa said.

Reply:
Changed